src/main/java/com/hivemq/client/internal/mqtt/handler/ssl/SslUtil.java

```diff
@@ -33,34 +33,20 @@
 
     private static final @NotNull String SSL_HANDLER_NAME = "ssl";
 
-    public static void initChannel(final @NotNull Channel channel, final @NotNull MqttClientSslConfigImpl sslConfig)
-            throws SSLException {
+    public static void initChannel(
+            final @NotNull Channel channel, final @NotNull MqttClientSslConfigImpl sslConfig,
+            final @NotNull String host, final int port) throws SSLException {
 
-        channel.pipeline().addFirst(SSL_HANDLER_NAME, createSslHandler(channel, sslConfig));
+        channel.pipeline().addFirst(SSL_HANDLER_NAME, createSslHandler(channel, sslConfig, host, port));
     }
 
     private static @NotNull SslHandler createSslHandler(
-            final @NotNull Channel channel, final @NotNull MqttClientSslConfigImpl sslConfig) throws SSLException {
-
-        final SSLEngine sslEngine = createSslEngine(channel, sslConfig);
-        final SslHandler sslHandler = new SslHandler(sslEngine);
-
-        sslHandler.setHandshakeTimeoutMillis(sslConfig.getHandshakeTimeoutMs());
-        return sslHandler;
-    }
-
-    static @NotNull SSLEngine createSslEngine(
-            final @NotNull Channel channel, final @NotNull MqttClientSslConfigImpl sslConfig) throws SSLException {
-
-        final SSLEngine sslEngine = createSslContext(sslConfig).newEngine(channel.alloc());
-
-        sslEngine.setUseClientMode(true);
-
-        return sslEngine;
+            final @NotNull Channel channel, final @NotNull MqttClientSslConfigImpl sslConfig,
+            final @NotNull String host, final int port) throws SSLException {
+        return createSslContext(sslConfig).newHandler(channel.alloc(), host, port);
```

Comment:
The javadoc of `newHandler` states that hostname verification is not enabled by default.
@dajudge Could you check this?
```
     * <p><b>Beware</b>: the underlying generated {@link SSLEngine} won't have
     * <a href="https://wiki.openssl.org/index.php/Hostname_validation">hostname verification</a> enabled by default.
     * If you create {@link SslHandler} for the client side and want proper security, we advice that you configure
     * the {@link SSLEngine} (see {@link javax.net.ssl.SSLParameters#setEndpointIdentificationAlgorithm(String)}):
```
https://docs.oracle.com/javase/8/docs/api/javax/net/ssl/SSLParameters.html#setEndpointIdentificationAlgorithm-java.lang.String-

Reply:
Nice catch! I believe however that this situation is present before the PR, as well. The modifications of this PR boil down to calling https://github.com/netty/netty/blob/8d9cea2ce0355c8ae907e164a72885216d1ba4e4/handler/src/main/java/io/netty/handler/ssl/SslContext.java#L943 which seems to be a similar same chain of invocations as happens without the PR. Or am I missing something here?

If the behavior regarding host name validation does not change with this PR I suggest to raise a separate issue for investigating the host name validation topic instead of blocking this PR with it. @SgtSilvio what's your take on this?

Reply:
Sure, I just asked in case you already know something about it.